frontend/src/v5/ui/routes/dashboard/projects/drawings/uploadDrawingRevisionForm/uploadList/uploadListItem/components/uploadListItemDestination/uploadListItemDestination.component.tsx

```diff
@@ -0,0 +1,232 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { memo, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { ErrorTooltip } from '@controls/errorTooltip';
+import { createFilterOptions } from '@mui/material';
+import { DrawingsHooksSelectors, ProjectsHooksSelectors } from '@/v5/services/selectorsHooks';
+import { canUploadToBackend, prepareSingleDrawingData } from '@/v5/store/drawings/drawings.helpers';
+import { formatMessage } from '@/v5/services/intl';
+import { Role } from '@/v5/store/currentUser/currentUser.types';
+import { isCollaboratorRole } from '@/v5/store/store.helpers';
+import { orderBy } from 'lodash';
+import { IDrawing } from '@/v5/store/drawings/drawings.types';
+import { name as drawingNameScheme } from '@/v5/validation/shared/validators';
+import { DestinationAutocomplete, DestinationInput, NewOrExisting, OptionsBox } from '@components/shared/uploadFiles/uploadList/uploadListItem/uploadListItemDestination/uploadListItemDestination.styles';
+import { AlreadyUsedName } from '@components/shared/uploadFiles/uploadList/uploadListItem/uploadListItemDestination/options/alreadyUsedName/alreadyUsedName.component';
+import { NewDestination } from '@components/shared/uploadFiles/uploadList/uploadListItem/uploadListItemDestination/options/newDestination/newDestination.component';
+import { NewDestinationInUse } from '@components/shared/uploadFiles/uploadList/uploadListItem/uploadListItemDestination/options/newDestinationInUse/newDestinationInUse.component';
+import { ExistingDestination } from '@components/shared/uploadFiles/uploadList/uploadListItem/uploadListItemDestination/options/existingDestination/existingDestination.component';
+
+const NO_OPTIONS_TEXT_ADMIN = formatMessage({
+	id: 'uploads.destination.noOptions.admin',
```

Comment:
There are few formatted messages on this page which have the exact same ids as those in the containers upload modal. I believe these have to be unique